pulser-core/pulser/sequence/sequence.py

```diff
@@ -825,6 +825,9 @@ def build(
                 >>> seq1 = seq.build(x=0.5, y=[1, 2, 3])
         """
         # Shallow copy with stored parametrized objects (if any)
+        # NOTE: While seq is a shallow copy, be extra careful with changes to
```

Comment:
Do we actually need those references or should we use `deepcopy` instead of `copy` to be safer ?

Reply:
A `deepcopy` before `Sequence._reset_parametrized()` triggers a bunch of recursiveness errors. However, I can (and will) do is run `copy` -> `_reset_parametrized` -> `deepcopy` to ensure nothing fishy happens while the copy is shallow.

Reply:
This seems to be already implemented actually

Reply:
Yes, I'll just place one after the other so that nothing is placed in between them in the fututre.